modules/zpca/zpca.py

```diff
@@ -146,20 +203,31 @@ def parse_zpca_logs(self, f: dict):
                 words.append(word)
                 word = []
 
-        # Concatinating each character
+        # Concatinating each word
         for k in words:
-            listToStr1.append("".join([str(elem) for elem in k]))
-        # Conactinating each word
-        for k in listToStr1:
             listToStr.append("".join([str(elem) for elem in k]))
 
+        len_listToStr = len(listToStr)
+        if len_listToStr == 36:
```

Comment:
I am not sure what is happening here, but is it safe to check based on the length of the list?

Reply:
I second that; this code fragment is similar to the one above for which I also raised concern. Could you please add some info in the comment here as well?

Reply:
Comments added!

Reply:
@krish8484 
These comments are not very descriptive... Could you please explain (in the comments) how does `len_listToStr` being `6` `8` or `36` correspond to the number of principal components: `1`, `2` `3`?
What is the logic and intention here? what do you check? why these exact numbers appear?